### DISCUSSION TITLE
RFC: retire TOML support in Architect 10?

### DISCUSSION THREAD

**Discussion by ryanblock** [architect/architect#1294] [General]
TOML support was added to Architect 6.0.28 in December 2019. The reasoning was, I believe, because Rust folks might appreciate our supporting it because Cargo uses TOML.

We prioritize backward compatibility with Architect with very few exceptions. I'd like to propose this is one of them. Rationale:
- TOML's restriction on mixed types in a collection means combining simple and verbose Arc resources isn't possible, as it is in other formats ([example here](https://arc.codes/docs/en/reference/project-manifest/events), click to the TOML tab)
- TOML adds a fair amount of bloat to the [Architect Parser](https://github.com/architect/parser/blob/main/src/compat/vendor/toml.min.js) (~60KB, or ~20% of its current 323KB)
- TOML accounts for ~10-20% of our filesystem existence hits upon each run of Architect Inventory
- TOML documentation has proven difficult to maintain and sometimes buggy with syntax highlighting
- Since its introduction, we have little to no evidence of its use; when recently polled in the [Architect Discord](https://discord.com/channels/880272256100601927/884128391341678742/900913804697808896), no one stood up to request we retain TOML support

I'd love feedback here; if we can't come up with a compelling rationale, I will mark TOML support for deprecation in [Arc 10](https://github.com/architect/architect/issues/1233).

**Comment by tbeseda** [comment]
From a documentation/support standpoint, this makes a lot sense.

I'd be curious in what environments users are preferring TOML over the the other options.

Also, because Arc configuration can be in JSON, theoretically a user could write TOML that then is built to `arc.json` right before it's needed.
I know plugins are coming out of beta, is there a hook available to accomplish something like this? It would have be before inventory is built.

**Reply by ryanblock** [reply]
A hook to use an arbitrary file format → JSON as a plugin would be dope. I can get into that. Let me think about if/how such a thing would be possible. Not sure if it is, but I do love to pave such paths during breaking changes.

**Comment by ryanblock** [comment]
Added this to the Arc 10 breaking change list: https://github.com/architect/architect/issues/1233
